infra/src/test/kotlin/fi/fta/geoviite/infra/ui/pagemodel/map/MapToolPanelElement.kt

```diff
@@ -13,13 +13,30 @@ class GeometryPlanGeneralInfoBox(by: By) : InfoBox(by) {
         Thread.sleep(500)
     }
 
+    fun huomiot(): String = fieldValue("Huomiot")
+    fun author(): String = fieldValue("Yritys")
     fun projekti(): String = fieldValue("Projekti")
-    fun luotu(): String = fieldValue("Luotu")
-    fun laatu(): String = fieldValue("Laatu")
+    fun tiedosto(): String = fieldValue("Tiedosto")
     fun vaihe(): String = fieldValue("Vaihe")
     fun vaiheenTarkennus(): String = fieldValue("Vaiheen tarkennus")
     fun ratanumero(): String = fieldValue("Ratanumero")
-    fun alkupKoordinaattijarstelma(): String = fieldValue("Alkuperäinen koordinaattijärjest.")
+    fun kilometriAlku(): String = fieldValue("Kilometri alku")
+    fun kilometriLoppu(): String = fieldValue("Kilometri loppu")
+}
+
+class GeometryPlanQualityInfobox(by: By) : InfoBox(by) {
+    init {
+        //Object initializes too quickly and webelement is not stable/ready
+        Thread.sleep(500)
+    }
```

Comment:
Rupesin googlettamaan että onko Seleniumissa mitään fiksumpaa tapaa odottaa kuin sleeppaaminen. Tuli vastaan sellainen kuin `WebDriverWait`, jolle voi myös antaa conditionin jonka täyttymiseen saakka se odottaa (https://stackoverflow.com/questions/59130200/selenium-wait-until-element-is-present-visible-and-interactable). Jos mahdollista, niin tätä voisi koittaa vähintään protoilla noiden ekplisiittisten Tread.sleep:ien käyttämisen sijaan.

Reply:
Joo taidan vastaisuudessa tehdä niin, että jos testikoodissa tehdään jotain huonosti, niin en ainakaan sentään kopioi huonosti tehtyä koodia vaikka sillä tavalla saisikin koodin senastisia kuvioita seurattua (esim. tässä se, että testikoodissa olisi valmiiksi accessorit noille kälikentille, vaikka niitä käyttäviä testejä ei varsinaisesti olekaan); eritoten siksi, koska käytännössä noita tosiaan on sitten oikeita testejä tehdessä joutunutkin puljaamaan ihan eri tavalla, esim. navigaatiopaneelin `waitForLocationTrackNamesTo`